src/graphnet/models/graphs/edges/minkowski.py

```diff
@@ -0,0 +1,98 @@
+"""Module containing EdgeDefinitions based on the Minkowski Metric."""
+from typing import Optional, List
+
+import torch
+from torch_geometric.data import Data
+from torch_geometric.utils import to_dense_batch
+from graphnet.models.graphs.edges.edges import EdgeDefinition
+
+
+def compute_minkowski_distance_mat(
+    x: torch.Tensor,
+    y: torch.Tensor,
+    c: float,
+    space_coords: Optional[List[int]] = None,
+    time_coord: Optional[int] = 3,
+) -> torch.Tensor:
+    """Compute all pairwise Minkowski distances.
+
+    Args:
+        x: First tensor of shape (n, d).
+        y: Second tensor of shape (m, d).
+        c: Speed of light, in scaled units.
+        space_coords: Indices of space coordinates.
+        time_coord: Index of time coordinate.
+
+    Returns: Matrix of shape (n, m) of all pairwise Minkowski distances.
+    """
+    space_coords = space_coords or [0, 1, 2]
```

Comment:
If I understand your question right, you're asking why I have `space_coords=None` in the function definition, and then later do `space_coords = space_coords or [0, 1, 2]`.

The reason is that I generally avoid having mutable objects as default parameters, since IMO python treats mutable default objects in a non-intuitive way, and it can create some hard-to-debug bugs.

An example from the [Python Documentation](https://docs.python.org/3/tutorial/controlflow.html#defining-functions) would be:
```python
def f(a, L=[]):
    L.append(a)
    return L

print(f(1))  # prints [1]
print(f(2))  # prints [1, 2]
print(f(3))  # prints [1, 2, 3]
```
The above behavior is related to how python treats function definitions. More information can be found in Section 4.8.1 and the related warning in the link-above.
The same behavior was what was changed in #633